client_vpn/main.tf

```diff
@@ -93,3 +114,39 @@ resource "aws_cloudwatch_log_group" "this" {
   retention_in_days = 14
   tags              = local.common_tags
 }
+
+#
+# Certificate based auth
+#
+
+resource "tls_private_key" "client_vpn" {
+  count     = var.authentication_option == "certificate-authentication" ? 1 : 0
+  algorithm = "RSA"
+}
+
+resource "tls_self_signed_cert" "client_vpn" {
+  count                 = var.authentication_option == "certificate-authentication" ? 1 : 0
+  private_key_pem       = tls_private_key.client_vpn[0].private_key_pem
+  validity_period_hours = 8760
+
+  subject {
+    common_name  = "cds-snc.ca"
+    organization = "Canadian Digital Service"
```

Comment:
I suspect we'll be the only one's using this module, but could you make these two options variables with these as the default values.